### PR TITLE
fix(lisong): fix icm/rnd+onppo config bugs and app_key env bugs

### PR DESCRIPTION
**Pull request by song2181**
## Description
fix config max_envstep. fix MiniGrid-AKTDT-7x7-1-v0 bug
## Check List

- [x] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/entry/serial_entry_reward_model_offpolicy.py [hunk 1]
```diff
@@ -103,7 +104,8 @@ def serial_pipeline_reward_model_offpolicy(
             replay_buffer.push(new_data, cur_collector_envstep=collector.envstep)
         # update reward_model
         reward_model.train()
-        reward_model.clear_data()
+        if count % cfg.reward_model.clear_buffer_per_iters == 0:
+            reward_model.clear_data()
```
Comment (PaParaZz1) [inline ding/entry/serial_entry_reward_model_offpolicy.py:108]: add comments for why clear data here

File: dizoo/minigrid/__init__.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from gym.envs.registration import register
+from gymnasium.envs.registration import register
```
Comment (PaParaZz1) [inline dizoo/minigrid/__init__.py:1]: why modify this here

File: dizoo/minigrid/config/minigrid_icm_offppo_config.py [hunk 3]
```diff
@@ -1,64 +1,83 @@
...
-minigrid_ppo_icm_create_config = dict(
+minigrid_icm_offppo_config = EasyDict(minigrid_icm_offppo_config)
+main_config = minigrid_icm_offppo_config
+minigrid_icm_offppo_create_config = dict(
     env=dict(
         type='minigrid',
         import_names=['dizoo.minigrid.envs.minigrid_env'],
     ),
-    env_manager=dict(type='subprocess'),
+    env_manager=dict(type='base'),
```
Comment (PaParaZz1) [inline dizoo/minigrid/config/minigrid_icm_offppo_config.py:73]: why `base` here

File: dizoo/minigrid/config/minigrid_icm_offppo_config.py [hunk 4]
```diff
@@ -1,64 +1,83 @@
...
             obs_shape=2835,
             action_shape=7,
+            action_space='discrete',
             encoder_hidden_size_list=[256, 128, 64, 64],
+            critic_head_hidden_size=64,
+            actor_head_hidden_size=64,
         ),
         learn=dict(
-            update_per_collect=10,
+            epoch_per_collect=10,
```
Comment (PaParaZz1) [inline dizoo/minigrid/config/minigrid_icm_offppo_config.py:47]: remove unused config fields like `epoch_per_collect` in offppo

File: dizoo/minigrid/envs/app_key_to_door_treasure.py [hunk 5]
```diff
@@ -185,7 +186,7 @@ def step(self, action):
 
         obs = self.gen_obs()
 
-        return obs, reward, done, {}
+        return obs, reward, done, done, {}
```
Comment (PaParaZz1) [inline dizoo/minigrid/envs/app_key_to_door_treasure.py:189]: why two `done` here

File: setup.py [hunk 6]
```diff
@@ -132,7 +132,7 @@
             'bsuite',
         ],
         'minigrid_env': [
-            'minigrid',
+            'minigrid==2.0.0',
```
Comment (PaParaZz1) [inline setup.py:135]: maybe >=?

File: ding/reward_model/rnd_reward_model.py [hunk 7]
```diff
@@ -60,19 +59,13 @@ class RndRewardModel(BaseRewardModel):
...
         obs_norm_clamp_min=-1,
         obs_norm_clamp_max=1,
-        intrinsic_reward_weight=None,
         # means the relative weight of RND intrinsic_reward.
-        # If intrinsic_reward_weight=None, we will automatically set it based on
-        # the absolute value of the difference between max and min extrinsic reward in the sampled mini-batch
-        # please refer to  estimate() method for details.
-        intrinsic_reward_rescale=0.01,
-        # means the rescale value of RND intrinsic_reward only used when intrinsic_reward_weight is None
+        intrinsic_reward_weight=0.01,
```
Comment (PaParaZz1) [inline ding/reward_model/rnd_reward_model.py:63]: add `extrinsic_reward_norm` and related keys  in default config

File: ding/reward_model/icm_reward_model.py [hunk 8]
```diff
@@ -207,17 +206,23 @@ def estimate(self, data: list) -> List[Dict]:
...
         with torch.no_grad():
             real_next_state_feature, pred_next_state_feature, _ = self.reward_model(states, next_states, actions)
-            reward = self.forward_mse(real_next_state_feature, pred_next_state_feature).mean(dim=1)
-            reward = (reward - reward.min()) / (reward.max() - reward.min() + 1e-8)
-            reward = reward.to(train_data_augmented[0]['reward'].device)
-            reward = torch.chunk(reward, reward.shape[0], dim=0)
-        for item, rew in zip(train_data_augmented, reward):
+            raw_icm_reward = self.forward_mse(real_next_state_feature, pred_next_state_feature).mean(dim=1)
+            icm_reward = (raw_icm_reward - raw_icm_reward.min()) / (raw_icm_reward.max() - raw_icm_reward.min() + 1e-8)
+            icm_reward = icm_reward.to(train_data_augmented[0]['reward'].device)
```
Comment (PaParaZz1) [inline ding/reward_model/icm_reward_model.py:211]: use `self.device` here

File: ding/reward_model/icm_reward_model.py [hunk 9]
```diff
@@ -207,17 +206,23 @@ def estimate(self, data: list) -> List[Dict]:
...
+            icm_reward = torch.chunk(icm_reward, icm_reward.shape[0], dim=0)
```
Comment (PaParaZz1) [inline ding/reward_model/icm_reward_model.py:212]: maybe we can omit this chunk operation here

File: ding/reward_model/icm_reward_model.py [hunk 10]
```diff
@@ -195,7 +195,6 @@ def _train(self) -> None:
     def train(self) -> None:
```
Comment (PaParaZz1) [inline ding/reward_model/icm_reward_model.py:195]: we should also add some tb items in ICM

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: dizoo/minigrid/envs/app_key_to_door_treasure.py (hunk 5 above)
Comment (song2181) [reply dizoo/minigrid/envs/app_key_to_door_treasure.py:189]: bacause in file minigrid_env.py , we use the step method of Gymnasium env, return is (observation, reward, terminated, truncated, info)
        obs, rew, done, _, info = self._env.step(action)

Comment (PaParaZz1) [reply dizoo/minigrid/envs/app_key_to_door_treasure.py:189]: add some comments about this

File: ding/reward_model/icm_reward_model.py [hunk 11]
```diff
@@ -207,17 +220,32 @@ def estimate(self, data: list) -> List[Dict]:
...
+            self.tb_logger.add_scalar('icm_reward/raw_icm_reward_max', raw_icm_reward.max(), self.estimate_cnt_icm)
+            self.tb_logger.add_scalar('icm_reward/raw_icm_reward_mean', raw_icm_reward.mean(), self.estimate_cnt_icm)
+            self.tb_logger.add_scalar('icm_reward/raw_icm_reward_min', raw_icm_reward.min(), self.estimate_cnt_icm)
+            self.tb_logger.add_scalar('icm_reward/raw_icm_reward_std', raw_icm_reward.std(), self.estimate_cnt_icm)
+            icm_reward = (raw_icm_reward - raw_icm_reward.min()) / (raw_icm_reward.max() - raw_icm_reward.min() + 1e-8)
+            self.tb_logger.add_scalar('icm_reward/icm_reward_max', icm_reward.max(), self.estimate_cnt_icm)
+            self.tb_logger.add_scalar('icm_reward/icm_reward_mean', icm_reward.mean(), self.estimate_cnt_icm)
+            self.tb_logger.add_scalar('icm_reward/icm_reward_min', icm_reward.min(), self.estimate_cnt_icm)
+            self.tb_logger.add_scalar('icm_reward/icm_reward_std', icm_reward.std(), self.estimate_cnt_icm)
+            icm_reward = (raw_icm_reward - raw_icm_reward.min()) / (raw_icm_reward.max() - raw_icm_reward.min() + 1e-8)
```
Comment (PaParaZz1) [inline ding/reward_model/icm_reward_model.py:234]: why norm twice here

File: dizoo/minigrid/__init__.py (hunk 2 above)
Comment (PaParaZz1) [reply dizoo/minigrid/__init__.py:1]: add this lib in setup.py

File: ding/reward_model/icm_reward_model.py [hunk 12]
```diff
@@ -171,8 +174,12 @@ def __init__(self, config: EasyDict, device: str, tb_logger: 'SummaryWriter') ->
         self.ce = nn.CrossEntropyLoss(reduction="mean")
         self.forward_mse = nn.MSELoss(reduction='none')
         self.reverse_scale = config.reverse_scale
+        self.res = nn.Softmax(dim=-1)
```
Comment (PaParaZz1) [inline ding/reward_model/icm_reward_model.py:177]: why use softmax here if we only need to sample action by argmax operation

File: dizoo/minigrid/config/minigrid_icm_offppo_config.py [hunk 13]
```diff
@@ -1,52 +1,69 @@
...
+            update_per_collect=1,
             batch_size=320,
-            learning_rate=0.0003,
+            learning_rate=3e-4,
             value_weight=0.5,
             entropy_weight=0.001,
             clip_ratio=0.2,
-            adv_norm=False,
+            adv_norm=True,
+            value_norm=True,
```
Comment (PaParaZz1) [inline dizoo/minigrid/config/minigrid_icm_offppo_config.py:53]: offppo doesn't have value norm

File: ding/reward_model/icm_reward_model.py [hunk 14]
```diff
@@ -151,6 +151,9 @@ class ICMRewardModel(BaseRewardModel):
         update_per_collect=100,
         # (float) the importance weight of the forward and reverse loss
         reverse_scale=1,
+        intrinsic_reward_weight=0.003,  # 1/300
```
Comment (PaParaZz1) [inline ding/reward_model/icm_reward_model.py:154]: add comments for each fields in default config

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]
